addons/product/models/product_product.py

```diff
@@ -397,6 +397,7 @@ def copy(self, default=None):
     def _search(self, args, offset=0, limit=None, order=None, count=False, access_rights_uid=None):
         # TDE FIXME: strange
         if self._context.get('search_default_categ_id'):
+            args = args.copy()
             args.append((('categ_id', 'child_of', self._context['search_default_categ_id'])))
```

Comment:
```suggestion
            args = args + [('categ_id', 'child_of', self._context['search_default_categ_id'])]
```
Better for the perf (I know, it's a detail, but still)
robodoo delegate=yelizariev